### DISCUSSION TITLE
WSGI wsgi.input out of spec

### DISCUSSION THREAD

**Discussion by cirospaciari** [emmett-framework/granian#23] [Show and tell]
Just to resume https://github.com/emmett-framework/granian/pull/22 discussion here:

@gi0baro actually the problem is HTTP_CONTENT_LENGTH should be CONTENT_LENGTH and HTTP_CONTENT_TYPE should be CONTENT_TYPE, i have this exactly same issue

@gi0baro another thing wsgi.input cannot be bytes, in my case I used BytesIO but is also wrong you can test with:

```python
from flask import Flask, make_response, request
from socketify import WSGI

app = Flask(__name__)

@app.route('/')
def index():
    """Test 6: Plaintext"""
    response = make_response(b"Hello, World!")
    response.content_type = "text/plain"
    return response

@app.route('/form', methods=['POST'])
def post_form():
    """Test 6: Plaintext"""
    user = request.form["name"]
    response = make_response(f"Hello, World {user}!")
    response.content_type = "text/plain"
    return response
@app.route('/json', methods=['POST'])
def post_json():
    """Test 6: Plaintext"""
    data = request.json
    response = make_response(f"Hello, World {data}!")
    response.content_type = "text/plain"
    return response

@app.route('/raw', methods=['POST'])
def post_raw():
    """Test 6: Plaintext"""
    data = request.data
    response = make_response(f"Hello, World {data}!")
    response.content_type = "text/plain"
    return response
```

I fixed socketify by taking a look at https://github.com/benoitc/gunicorn/blob/2f9eb19b666868d324150072e7489788688c91cb/gunicorn/http/body.py

and I come up with https://github.com/cirospaciari/socketify.py/blob/392af9784d6a94403e58d2c02652aedc3379bf4c/src/socketify/wsgi.py#L25

**Comment by gi0baro** [comment]
This is now tracked in #24
